renderer/pages/progress.js

```diff
@@ -0,0 +1,86 @@
+import { ipcRenderer } from "electron";
+import { fileSizeReadable } from "../utils";
+import Layout from "../components/Layout";
+import Logo from "../components/Logo";
+
+class Progress extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      transferred: 0,
+      percent: 0
+    };
+  }
+
+  componentDidMount() {
+    ipcRenderer.on("progress", (event, progressObject) => {
+      this.setState(progressObject);
```

Comment:
Don't set `progressObject` in your state directly. You only care about certain values in the `progressObject` object.

A better approach would be:

```
ipcRenderer.on("progress", (event, { transferred, total, percent }) => {
    this.setState({
        percent,
        total,
        transferred,
   });
});
```